### ISSUE TITLE
TypeError: Expected maxsize to be an integer or None

### ISSUE THREAD

**Issue by nickponline** [brycedrennan/imaginAIry#189]
Any command I run gives an error:

```
(dev) (master ✗)]$ imagine "colorful smoke" --steps 40 --upscale
NOTE: Redirects are currently not supported in Windows or MacOs.
╭─────────────────────────────── Traceback (most recent call last) ────────────────────────────────╮
│                                                                                                  │
│ /Users/me/opt/anaconda3/envs/dev/bin/imagine:5 in <module>                                    │
│                                                                                                  │
│   2 # -*- coding: utf-8 -*-                                                                      │
│   3 import re                                                                                    │
│   4 import sys                                                                                   │
│ ❱ 5 from imaginairy.cmds import imagine_cmd                                                      │
│   6 if __name__ == '__main__':                                                                   │
│   7 │   sys.argv[0] = re.sub(r'(-script\.pyw|\.exe)?$', '', sys.argv[0])                         │
│   8 │   sys.exit(imagine_cmd())                                                                  │
│ /Users/me/opt/anaconda3/envs/dev/lib/python3.7/site-packages/imaginairy/__init__.py:7 in      │
│ <module>                                                                                         │
│                                                                                                  │
│    4 os.putenv("PYTORCH_ENABLE_MPS_FALLBACK", "1")                                               │
│    5 import PIL.Image  # noqa                                                                    │
│    6                                                                                             │
│ ❱  7 from .api import imagine, imagine_image_files  # noqa                                       │
│    8 from .enhancers.describe_image_blip import generate_caption  # noqa                         │
│    9 from .schema import (  # noqa                                                               │
│   10 │   ImaginePrompt,                                                                          │
│                                                                                                  │
│ /Users/me/opt/anaconda3/envs/dev/lib/python3.7/site-packages/imaginairy/api.py:12 in <module> │
│                                                                                                  │
│     9 from PIL import Image, ImageDraw, ImageFilter, ImageOps                                    │
│    10 from pytorch_lightning import seed_everything                                              │
│    11                                                                                            │
│ ❱  12 from imaginairy.enhancers.clip_masking import get_img_mask                                 │
│    13 from imaginairy.enhancers.describe_image_blip import generate_caption                      │
│    14 from imaginairy.enhancers.face_restoration_codeformer import enhance_faces                 │
│    15 from imaginairy.enhancers.upscale_realesrgan import upscale_image                          │
│                                                                                                  │
│ /Users/me/opt/anaconda3/envs/dev/lib/python3.7/site-packages/imaginairy/enhancers/clip_maskin │
│ g.py:10 in <module>                                                                              │
│                                                                                                  │
│     7 import torch                                                                               │
│     8 from torchvision import transforms                                                         │
│     9                                                                                            │
│ ❱  10 from imaginairy.img_utils import pillow_fit_image_within                                   │
│    11 from imaginairy.log_utils import log_img                                                   │
│    12 from imaginairy.vendored.clipseg import CLIPDensePredT                                     │
│    13                                                                                            │
│                                                                                                  │
│ /Users/me/opt/anaconda3/envs/dev/lib/python3.7/site-packages/imaginairy/img_utils.py:9 in     │
│ <module>                                                                                         │
│                                                                                                  │
│    6 from einops import rearrange, repeat                                                        │
│    7 from PIL import Image                                                                       │
│    8                                                                                             │
│ ❱  9 from imaginairy.utils import get_device                                                     │
│   10                                                                                             │
│   11                                                                                             │
│   12 def pillow_fit_image_within(                                                                │
│                                                                                                  │
│ /Users/me/opt/anaconda3/envs/dev/lib/python3.7/site-packages/imaginairy/utils.py:17 in        │
│ <module>                                                                                         │
│                                                                                                  │
│    14                                                                                            │
│    15                                                                                            │
│    16 @lru_cache                                                                                 │
│ ❱  17 def get_device() -> str:                                                                   │
│    18 │   """Return the best torch backend available."""                                         │
│    19 │   if torch.cuda.is_available():                                                          │
│    20 │   │   return "cuda"                                                                      │
│                                                                                                  │
│ /Users/me/opt/anaconda3/envs/dev/lib/python3.7/functools.py:490 in lru_cache                  │
│                                                                                                  │
│   487 │   │   if maxsize < 0:                                                                    │
│   488 │   │   │   maxsize = 0                                                                    │
│   489 │   elif maxsize is not None:                                                              │
│ ❱ 490 │   │   raise TypeError('Expected maxsize to be an integer or None')                       │
│   491 │                                                                                          │
│   492 │   def decorating_function(user_function):                                                │
│   493 │   │   wrapper = _lru_cache_wrapper(user_function, maxsize, typed, _CacheInfo)            │
╰──────────────────────────────────────────────────────────────────────────────────────────────────╯
TypeError: Expected maxsize to be an integer or None
```

**Comment by nickponline** [comment]
I wonder if this it due to not having a GPU?

**Comment by brycedrennan** [comment]
This is because you're using Python 3.7.  I'll fix this particular issue but python 3.7 is end-of-life in 5 months. Would probably be good for you to update.

**Comment by brycedrennan** [comment]
Fixed in just released 8.0.3
